src/Printer.php

```diff
@@ -87,16 +87,22 @@ class Printer
 
     private $imposedScale;
 
+    private $forceWidth;
+
+    private $barcodeLabel;
+
     /**
      * @param integer $width
      * @param integer $height
      * @param null|string $fontPath
      */
-    public function __construct($width, $height, $fontPath = 'FreeSans.ttf')
+    public function __construct($width, $height, $forceWidth, $barcodeLabel, $fontPath = 'FreeSans.ttf')

```

Comment:
A mentioned above, printer should get calculated width and dont bother how the number been achieved. Barcode label belongs more to output() method.